pkg/client/results/reader.go

```diff
@@ -402,7 +481,26 @@ func (r *Reader) FileReader(filename string) (io.Reader, error) {
 	return returnReader, nil
 }
 
-// Return the full path of the ClusterHealthFile
+// ClusterHealthFilePath returns the full path of the ClusterHealthFile
 func ClusterHealthFilePath() string {
-	return  path.Join(metadataDir, ClusterHealthFile)
+	return path.Join(metadataDir, ClusterHealthFile)
+}
+
+// fileInfoToReader takes the given FileInfo object and tries to return a reader
+// for the data. In the case of normal FileInfo objects (e.g. from os.Stat())
+// you need to provide the full path to the file so it can be opened since the
+// FileInfo object only contains the name but not the directory.
+func fileInfoToReader(info os.FileInfo, path string) (io.Reader, error) {
+	switch v := info.Sys().(type) {
+	case io.Reader:
+		return info.Sys().(io.Reader), nil
+	case syscall.Stat_t, *syscall.Stat_t:
```

Comment:
It is frustrating; the docs says `.Sys()` should give you the backing data, but its just not the case for general FileInfo types. In this case you need to just call Open manually.